resources/Microsoft.Windows.Developer/Microsoft.Windows.Developer.psm1

```diff
@@ -800,6 +805,64 @@ class WindowsCapability {
         }
     }
 }
+
+[DSCResource()]
+class AdvancedNetworkSharingSetting {
+    [DscProperty(Key, Mandatory)]
+    [AdvancedNetworkSharingSettingName]$Name
+
+    [DscProperty()]
+    [string[]]$Profiles = @()
```

Comment:
This is tricky as it intends to disable any enabled profiles if they are not explicitly listed. Can anyone think of a better calling pattern for this? FWIW this is an [unresolved issue on the networking DSC provider](https://github.com/dsccommunity/NetworkingDsc/issues/356).

Reply:
Based upon the refactor involving Set-NetFirewallRule I believe that this is an acceptable pattern to start with as it will facilitate setting toggling if needed.